### ISSUE TITLE
Non uniform rectangular grid on Canvas Layer?

### ISSUE THREAD

**Issue by tylertucker202** [IHCantabria/Leaflet.CanvasLayer.Field#54]
Hello IHCatabria! Thanks again for supporting this outstanding library. I have been using a forked version of this library for the app [Argovis](https://argovis.colorado.edu/ng/grid) for close to a year now. We are collaborating with some oceanographers who made some gridded products, and we want to share them on Argovis using CanvasLayer.Field. The issue is that the grids are not on a uniformly spaced grid! Instead, their latitude gets finer at the south pole (5-degree polynomial!). Is there a way to make a canvas layer using a non-uniform grid using CanvasLayer?

FYI, I'm loading the gridded data as JSON from a database, for example, we have sea-ice coverage in a small box [here](https://argovis.colorado.edu/griddedProducts/nonUniformGrid/window?gridName=sose_si_area_3_day&presLevel=0&date=2013-01-04&latRange=[-55,-50]&lonRange=[0,%205])

Basically the grid data is comprised of an array of {lat: number, lon: number, value: number} objects. One solution I'm working on re-interpolates this grid on the front-end to something uniform. A non-trivial task...Before I work on this any further, I was wondering if there is a better way? I'm not as familiar with d3 to homebrew my own solution yet but there has to be a way for the d3/webGL lower level stuff to handle interpolation on a non-uniform grid. 

Your thoughts, suggestions, counsel, would be appreciated. I'm willing to code the solution myself and share it with the community. All I am asking is a nudge in the right direction. Thank you again for your time!

**Comment by Pedrazl** [comment]
Hi @tylertucker202 ,
i am sorry but actually the plugin only works with regular grids. I don´t know your specific use case, but you may try to interpolate your irregular grid with GDAL (or any other library that accomplish that transformation) before the visualization.

**Comment by tylertucker202** [comment]
Its OK. We managed to take regrid a sparse irregularly spaced grid [here](https://argovis.colorado.edu/ng/grid?presLevel=0&date=2013-01-01&shapes=%5B%5B9.667969,-73.627789,39.550781,-51.508742%5D%5D&gridName=sose_si_area_1_day_sparse&interpolateBool=false&colorScale=ice&inverseColorScale=false&paramMode=false&param=SIarea&gridDomain=%5B0,0.965%5D). Then we pass that to Canvas.Layer. What is nice is that you can store the grid as sparse and do the regridding on the front end.
